buffer/src/main/java/io/netty/buffer/PoolSubpage.java

```diff
@@ -115,7 +115,9 @@ boolean free(PoolSubpage<T> head, int bitmapIdx) {
 
         if (numAvail ++ == 0) {
             addToPool(head);
-            return true;
+            if (maxNumElems > 1) {
+                return true;
+            }
```

Comment:
The test you added passes without this change. So this is not really being tested.

Reply:
Has been adjusted.